src/core/ext/transport/chttp2/transport/flow_control.c

```diff
@@ -342,9 +375,52 @@ void grpc_chttp2_flowctl_destroy_stream(grpc_chttp2_transport_flowctl* tfc,
   announced_window_delta_preupdate(tfc, sfc);
 }
 
+// Returns an urgency with which to make an update
+static grpc_chttp2_flowctl_urgency delta_is_significant(
+    const grpc_chttp2_transport_flowctl* tfc, int32_t value,
+    grpc_chttp2_setting_id setting_id) {
+  int64_t delta = (int64_t)value -
+                  (int64_t)tfc->t->settings[GRPC_LOCAL_SETTINGS][setting_id];
+  if (delta != 0 && (delta <= -value / 20 || delta >= value / 20)) {
+    return GRPC_CHTTP2_FLOWCTL_QUEUE_UPDATE;
+  } else if (delta != 0 && (delta <= -value / 5 || delta >= value / 5)) {
+    return GRPC_CHTTP2_FLOWCTL_QUEUE_UPDATE;
```

Comment:
Do these two cases intentionally return the same value (despite one subsuming the other)? Is this case reserved for future expansion? If so, can you mark a TODO (preferably with a tracking bug on the feature)